replay2/memory_remapper/memory_remapper.cc

```diff
@@ -14,4 +14,98 @@
 
 #include "memory_remapper.h"
 
-void foo(int var) { return; }
+namespace agi {
+namespace replay2 {
+
+	ReplayAddress MemoryRemapper::AddMapping(const MemoryObservation& observation)
+	{
+		const CaptureAddress captureAddress = observation.captureAddress();
+
+		auto iter = CaptureAddressRangeIter(captureAddress);
+		if(iter != captureAddressRanges_.end()) {
+
+			const CaptureAddressRange captureAddressRange = iter->first;
+
+			const intptr_t offset = captureAddress.charPtr() -captureAddressRange.baseAddress().charPtr();
+			if(offset < captureAddressRange.length()) {
+				throw AddressAlreadyMappedException();
+			}
+		}
+
+		const size_t mappingLength = observation.resourceGenerator()->length();
+
+		char* replayAllocation = new char [mappingLength];
```

Comment:
Yeah, I though about this, but using smart pointers requires us to maintain another container for them. This might be one of the few exceptions where it's actually neater to use new/delete. This is low level memory code so it might be forgiveable